pallets/portfolio/src/lib.rs

```diff
@@ -205,8 +216,11 @@ decl_module! {
         /// * `PortfolioNotEmpty` if the portfolio still holds any asset
         #[weight = 1_000_000_000]
         pub fn delete_portfolio(origin, num: PortfolioNumber) -> DispatchResult {
-            let sender = ensure_signed(origin)?;
-            let did = Context::current_identity_or::<Identity<T>>(&sender)?;
+            let PermissionedCallOriginData {
+                primary_did,
+                secondary_key,
+                ..
+            } = Identity::<T>::ensure_origin_call_permissions(origin)?;
```

Comment:
Should we check that if the caller is a secondary_key, that they have permission to the portfolio being deleted?